python/guide/packages/gluon/trainer.md

```diff
@@ -1,72 +1,95 @@
 # Trainer
 
-A trainer updates neural network parameters by an optimization
-method.
+Training a neural network model consists of iteratively performing three simple steps. 
 
-## Basic Usages
+The first step is the forward step which computes the loss.  In MXNet gluon, this first step is achieved by doing a forward pass by calling `net.forward(X)` or simply `net(X)` and then calling the loss function with the result of the forward pass and the labels. For example `l = loss_fn(net(X), y)`.
 
-We first create a simple Perceptron and run forward and
-backward passes to obtain the gradients.
+The second step is the backward step which computes the gradient of the loss with respect to the parameters. In gluon, this step is  achieved by doing the first step in an `autograd.record()` scope to record the computations needed to calculate the loss, and then calling `l.backward()` to compute the gradient of the loss with respect to the parameters.
+
+The final step is to update the neural network model parameters using an optimization algorithm. In gluon this step is performed by the [`gluon.Trainer` API](http://beta.mxnet.io/api/gluon/mxnet.gluon.Trainer.html) and is the subject of this guide. The gluon trainer is instantiated with a set of learnable parameters and the optimization algorithm to learn those parameters and every training iteration, `trainer.step` updates the model parameters.
```

Comment:
remove word API

Reply:
gluon trainer -> `gluon.Trainer`

Reply:
Would reword last sentence: When creating a `gluon.Trainer` you must provide a collection of parameters that need to be learnt. You also provide an `Optimizer` that will be used to update the parameters every training iteration when `trainer.step` is called.

Reply:
addressed in next commit. resolving comment.

Reply:
thanks. changed and addressed.